### PR TITLE
SOF 170 - WP distance overflow

### PR DESCRIPTION
**Pull request by LiamFly**
Changed calculation of WP_DIST to use target coordinates, so that it can be be calculated without overflow issues

### CODE REVIEW

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 1]
```diff
@@ -849,8 +852,27 @@ public double glide_ratio
         [DisplayText("Dist to WP (dist)")]
         public float wp_dist
         {
-            get => _wpdist * multiplierdist;
-            set => _wpdist = value;
+            get
+            {
+                //Flylogix
+                if (lat == 0 && lng == 0 || next_lat == 0)
+                    return 0;
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:859]: return _wpdist * multiplierdist;

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 2]
```diff
@@ -849,8 +852,27 @@ public double glide_ratio
         [DisplayText("Dist to WP (dist)")]
         public float wp_dist
         {
-            get => _wpdist * multiplierdist;
-            set => _wpdist = value;
+            get
+            {
+                //Flylogix
+                if (lat == 0 && lng == 0 || next_lat == 0)
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:858]: if (_wpdist <  65535 || lat == 0 && lng == 0 || next_lat ==0)

File: ExtLibs/ArduPilot/CurrentState.cs (hunk 1 above)
Comment (rob-flylogix) [reply ExtLibs/ArduPilot/CurrentState.cs:859]: i.e. we return the existing method if _wpdist < 65535 or lat or lng == 0 or next_lat == 0

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 3]
```diff
@@ -1497,7 +1519,7 @@ public float GeoFenceDist
             }
         }
         */
-        [GroupText("Position")]
+                [GroupText("Position")]
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:1522]: remove extra tab

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 4]
```diff
@@ -1954,6 +1976,8 @@ private void Parent_OnPacketReceived(object sender, MAVLink.MAVLinkMessage mavLi
 
                     {
                         var postraget = mavLinkMessage.ToStructure<MAVLink.mavlink_position_target_global_int_t>();
+                            next_lat = postraget.lat_int / 1e7;
+                            next_lon = postraget.lon_int / 1e7;
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:1981]: remove extra tab indents

Comment (rob-flylogix) [reply ExtLibs/ArduPilot/CurrentState.cs:1981]: correct spelling of postarget

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 5]
```diff
@@ -2697,6 +2721,8 @@ private void Parent_OnPacketReceived(object sender, MAVLink.MAVLinkMessage mavLi
                     }
 
                         break;
+
+
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:2725]: remove extra newlines

**Review by rob-flylogix — CHANGES_REQUESTED** [review]

File: ExtLibs/ArduPilot/CurrentState.cs (hunk 4 above)
Comment (LiamFly) [reply ExtLibs/ArduPilot/CurrentState.cs:1981]: > correct spelling of postarget

This is how it was spelt in the original function, but I can correct the rest of it if you want

Comment (rob-flylogix) [reply ExtLibs/ArduPilot/CurrentState.cs:1981]: yes please - I've been tidying up as I find things like this

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 6]
```diff
@@ -854,9 +854,9 @@ public float wp_dist
         {
             get
             {
+                if (_wpdist < 65535 || lat == 0 && lng == 0 || next_lat == 0)
+                    return _wpdist * multiplierdist;
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:858]: I think we should add "&& next_lng == 0" to next_lat check:

if (_wpdist < 65535 || lat == 0 && lng == 0 || next_lat == 0 && next_lng == 0)

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 7]
```diff
@@ -990,9 +992,9 @@ public double tob
                 DateTime currentTime = DateTime.UtcNow;
                 int tobHours = currentTime.Hour + (int)(tobSeconds / 3600);
                 int tobMinutes = currentTime.Minute + (int)(tobSeconds / 60);
-                if (tobMinutes >= 60)
+                while (tobMinutes >= 60)
                 {
-                    tobHours += 1;
+                    //tobHours += 1;
                     tobMinutes -= 60;
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:998]: Remove commented out line ?

**Review by rob-flylogix — CHANGES_REQUESTED** [review]
A couple of small things

**Review by rob-flylogix — APPROVED** [review]

**Pull request closed without merging**